elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,941 @@ function xml_form_elements_creative_commons($element, &$form_state) {
+      }
+      elseif ($response->getName() == 'error') {
+        drupal_set_message(t("The Creative Commons API unexpectedly returned an error. Creating license URI manually."), 'warning');
+      }
+      $new_license_uri = xml_form_elements_creative_commons_get_uri_from_values($values);
+    }
+    // Store the new license URI.
+    $values['license_fieldset']['license_uri'] = $new_license_uri;
+  }
+
+  // If there was an error, return.
+  if ($error) {
+    return $values;
+  }
+
+  // Check the URI we are about to return.
+  if (!empty($values['license_fieldset']['license_uri'])) {
+    $current_query = xml_form_elements_creative_commons_make_value_array('', '', '', '', '', $values['license_fieldset']['license_uri']);
+    $response = xml_form_elements_creative_commons_retrieve_api_response($element['#name'], 'api_data', $current_query, $form_state);
+    if ($response) {
+      if ($response->getName() == 'error') {
+        // The license URI is not recognized.
+        $error = TRUE;
+        if ($values['license_fieldset']['license_type'] == 'manual') {
+          form_set_error($element['#name'] . '][license_fieldset][license_uri', t("CC license URI seems to be invalid. Perhaps you would prefer to choose one automatically?"));
+        }
+        else {
+          form_set_error($element['#name'] . '][license_fieldset][license_version', t("CC license version does not seem to exist."));
+        }
+      }
+    }
+    else {
+      // We could not reach the API.
+      drupal_set_message(t("Could not reach Creative Commons API to validate your input."), 'warning');
+    }
+  }
+
+  // Return the values for the form element.
+  return $values;
+}
+
+/**
+ * Get API path for license.
+ *
+ * @param string $license
+ *   A string containing the license ('zero', 'mark', 'by' or 'sampling+').
+ *
+ * @return string|FALSE
+ *   The path where the license can be retrieved from using the API
+ *   ('zero', 'mark', 'recombo' or 'standard'), or FALSE if $license was
+ *   not recognized.
+ */
+function xml_form_elements_creative_commons_get_api_path($license) {
+  $zero = array('zero');
+  $mark = array('mark');
+  $recombo = array('sampling+');
+  $standard = array('by');
+  $subpath = '';
+  if (in_array($license, $zero)) {
+    $subpath = 'zero';
+  }
+  elseif (in_array($license, $mark)) {
+    $subpath = 'mark';
+  }
+  elseif (in_array($license, $recombo)) {
+    $subpath = 'recombo';
+  }
+  elseif (in_array($license, $standard)) {
+    $subpath = 'standard';
+  }
+  else {
+    return FALSE;
+  }
+  return 'license/' . $subpath . '/get';
+}
+
+/**
+ * Get a license from the CC API.
+ *
+ * @param string $path
+ *   The path from which to query.
+ * @param array $options
+ *   The http query array to attach to the query.
+ *
+ * @return SimpleXMLElement|FALSE
+ *   The response from the REST API as a SimpleXMLElement, or FALSE if request
+ *   failed.
+ *
+ * Note that there may be a bug in the CC API. Querying, e.g., 'fi'
```

Comment:
@DiegoPino Excellent point! Where in the doc should it go? WIKI/Readme/...? If outside this repo, could you take care of that, please?

Reply:
I would say in the XML forms wiki , github, and a note in the README.md (known issues?) @rosiel ?

Reply:
@DiegoPino @rosiel If you want, I can write a note (maybe an item in a new sub-section `### Known Issues` under `## Troubleshooting/Issues`?) in the `README.md`. On the other hand, this is actually outside the scope of this particular PR, so it could also be done later (in case we are tight on time before release)...